packages/multi-select-combo-box/test/basic.test.js

```diff
@@ -878,48 +878,51 @@ describe('basic', () => {
       inputElement.click();
       expect(internal.opened).to.be.false;
     });
+  });
 
-    describe('dataProvider', () => {
-      let spyAsyncDataProvider;
+  describe('readonly + dataProvider', () => {
+    let spyAsyncDataProvider;
 
-      const ITEMS = ['apple', 'banana', 'lemon', 'orange', 'pear'];
+    const ITEMS = ['apple', 'banana', 'lemon', 'orange', 'pear'];
 
-      const getDataProvider = (allItems) => (params, callback) => {
-        const filteredItems = allItems.filter((item) => item.indexOf(params.filter) > -1);
-        const size = filteredItems.length;
-        const offset = params.page * params.pageSize;
-        const dataProviderItems = filteredItems.slice(offset, offset + params.pageSize);
-        callback(dataProviderItems, size);
-      };
+    const getDataProvider = (allItems) => (params, callback) => {
+      const filteredItems = allItems.filter((item) => item.indexOf(params.filter) > -1);
+      const size = filteredItems.length;
+      const offset = params.page * params.pageSize;
+      const dataProviderItems = filteredItems.slice(offset, offset + params.pageSize);
+      callback(dataProviderItems, size);
+    };
 
-      const dataProvider = getDataProvider(ITEMS);
+    const dataProvider = getDataProvider(ITEMS);
 
-      const asyncDataProvider = (params, callback) => {
-        setTimeout(() => dataProvider(params, callback));
-      };
+    const asyncDataProvider = (params, callback) => {
+      setTimeout(() => dataProvider(params, callback));
+    };
 
-      beforeEach(() => {
-        comboBox.items = undefined;
-        spyAsyncDataProvider = sinon.spy(asyncDataProvider);
-        comboBox.dataProvider = spyAsyncDataProvider;
-      });
+    beforeEach(() => {
+      comboBox.items = undefined;
```

Comment:
The current solution works fine with either a data provider or an items array but it doesn't support changing one in favor of another with `selectedItems` + `readonly` set before. To my understanding, this is not a common case, so I decided to just adjust the tests.

Reply:
Changing the `items` / `dataProvider` while `readonly` is not a common case. So the change looks good to me.